### PR TITLE
mxnet: 1.8.0 -> 1.9.1

### PR DESCRIPTION
**Pull request by tjni**
###### Description of changes

https://github.com/apache/incubator-mxnet/releases/tag/1.9.1
https://github.com/apache/incubator-mxnet/releases/tag/1.9.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/mxnet/default.nix [hunk 1]
```diff
@@ -13,25 +13,26 @@ assert cudnnSupport -> cudaSupport;
...
 
   patches = [
-    # Fix build error https://github.com/apache/incubator-mxnet/issues/19405
+    # Remove the following two patches when updating mxnet to 2.0.
     (fetchpatch {
-      name = "mxnet-fix-gcc-linker-error-1.patch";
-      url = "https://github.com/apache/incubator-mxnet/commit/78e31d66d19e385ca4ef73245ce79a47e375d8d1.diff";
-      sha256 = "sha256-UfmGhh4RbvrEOXe6IJxHm1Aqpy1gS6gHxrX5KQBXjv4=";
+      name = "1-auto-disable-sse-for-non-x86.patch";
+      url = "https://github.com/apache/incubator-mxnet/commit/55e69871d4cadec51a8bbb6700131065388cb0b9.patch";
```
Comment (risicle) [inline pkgs/applications/science/math/mxnet/default.nix:28]: Does this commit have a variant that exists on an upstream branch? If not does it belong to an unmerged PR you could link to?

Comment (risicle) [reply pkgs/applications/science/math/mxnet/default.nix:28]: Sorry, I mean the other patch :point_down:

Comment (tjni) [reply pkgs/applications/science/math/mxnet/default.nix:28]: Found it and updated the PR. Thanks for pointing it out.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64

**Review by risicle — APPROVED** [review]

**Pull request merged**